troi/patches/periodic_jams.py

```diff
@@ -15,27 +15,35 @@
 MAX_RECS_LIMIT = 1000  # the maximum of recommendations available in LB
 
 
-class DailyJamsPatch(troi.patch.Patch):
+class PeriodicJamsPatch(troi.patch.Patch):
     """
-       Take the raw recommendations, filter out recently listened tracks, unlisted tracks, hated tracks
-       and then randomly pick 50 of them, with never more than 2 recordings by the same artist.
+       Create either daily-jams, weekly-jams or weekly-new-jams with this patch.
+
+       First, fetch the top recommendations. For daily-jams and weekly-jams, filter out the recently listened tracks.
+       For weekly-new-jams, filter out tracks that have been listened to.
+
+       Then filter out hated tracks and make the playlist.
     """
 
+    JAM_TYPES = ("daily-jams", "weekly-jams", "weekly-new-jams")
+
     def __init__(self, debug=False):
         super().__init__(debug)
 
     @staticmethod
     def inputs():
         """
-        Generate a daily playlist from the ListenBrainz recommended recordings.
+        Generate a periodic playlist from the ListenBrainz recommended recordings.
 
         \b
         USER_NAME is a MusicBrainz user name that has an account on ListenBrainz.
+        TYPE Must be one of "daily-jams", "weekly-jams" or "weekly-new-jams".
         JAM_DATE is the date for which the jam is created (this is needed to account for the fact different timezones
         can be on different dates). Required formatting for the date is 'YYYY-MM-DD'.
         """
         return [
             {"type": "argument", "args": ["user_name"]},
+            {"type": "argument", "args": ["type"], "kwargs": {"required": False}},
```

Comment:
I see the shortcoming of the way we generate commands now. We can't pass a click.Choice here without importing click. :disappointed: